deltacat/utils/metrics.py

```diff
@@ -0,0 +1,111 @@
+import ray
+import logging
+
+from deltacat import logs
+from enum import Enum
+from typing import Dict, Any, List, Callable
+from deltacat.aws.clients import resource_cache
+from datetime import datetime
+
+from botocore.exceptions import ClientError
+from ray._private.services import get_node_ip_address
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+DEFAULT_DELTACAT_METRICS_NAMESPACE = "ray-deltacat-profiling"
+
+
+class MetricsConfig:
+    def __init__(self, region: str, job_run_id: str):
+        self.region = region
+        self.job_run_id = job_run_id
+
+
+class MetricsType(str, Enum):
```

Comment:
You may probably need `MetricsTarget` and set that to `CloudWatch` so that it is generic for uses across cloud providers.

Reply:
Added MetricsTarget